invoice_margin/hooks.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2017 - Today: GRAP (http://www.grap.coop)
+# @author: Sylvain LE GAL (https://twitter.com/legalsylvain)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+def _create_column(cr, table_name, column_name, column_type):
+    req = "ALTER TABLE %s ADD COLUMN %s %s" % (
+        table_name, column_name, column_type)
+    cr.execute(req)
+
+
+def pre_init_hook(cr):
+    _logger.info(
+        "Compute account_invoice_line.purchase_price for existing lines")
```

Comment:
> The creation of a new computed field consider re-compute the existing lines currently.
So this code is not longer required.

purchase_price is not a computed field. (like in #292). So it has to be prepopulated by SQL script.
https://github.com/OCA/account-invoicing/pull/308/files#diff-a08017a90fbce7fdbe067b22197b76f9R24